src/MongoDB.Driver.Core/Core/WireProtocol/CommandUsingCommandMessageWireProtocol.cs

```diff
@@ -75,9 +86,17 @@ public TCommandResult Execute(IConnection connection, CancellationToken cancella
             connection.SendMessage(message, _messageEncoderSettings, cancellationToken);
             MessageWasSent();
 
-            var encoderSelector = new CommandResponseMessageEncoderSelector();
-            var response = (CommandResponseMessage)connection.ReceiveMessage(message.RequestId, encoderSelector, _messageEncoderSettings, cancellationToken);
-            return ProcessResponse(connection.ConnectionId, response.WrappedMessage);
+            var wrappedMessage = message.WrappedMessage;
+            if (wrappedMessage.ResponseExpected)
```

Comment:
The value of ResponseExpected (which is computed from the MoreToCome flag) might change after the message was first created depending on what the PostWriteAction does.

I.e., it might be false initially for { w : 0 } and become true for ordered batches other than the last.